### DISCUSSION TITLE
Bootloop but phone actually has booted

### DISCUSSION THREAD

**Discussion by SoldierDragon** [LSPosed/LSPosed#2516] [Q&A]
Hi guys, i installed LSPosed and the module XDowngrader, and manage to get it working, but as soon i restarted my phone again it didn't boot anymore, from what I could understand it enters the OS, because I can briefly see the adb device and enter the adb shell, but then the phone restarts, I got the last_kmsg file, anyone might have an idea of what's causing this?
Already tried to remove the LSPosed from /data/adb/modules and i removed the /data/adb/lspd
I had the Magisk v24.3 and the latest LSPosed v1.8.6
Below my last_kmsg
[last_kmsg.txt](https://github.com/LSPosed/LSPosed/files/11369673/last_kmsg.txt)

**Comment by viki3d** [comment]
What Custom-ROM you are using?

A case when OS boots but bootloop occurs is when dm-verity protection triggers.
If this is your case: You can boot in Recovery mode (Restart and hold Pwr+VolumeUp) and re-flash you ROM again.

Now checking your "last_kmsg.txt": You are with Samsung ROM with custom recovery.
You can see - your vbmeta partition hash sums don't match:

_[0:       1.833] avb_slot_verify.c:966: DEBUG: Loading vbmeta struct from partition 'vbmeta'.
avb_vbmeta_image.c:212: ERROR: Hash does not match!_

Maybe your problem is from XDowngrader, which modifier some system partitions.
You can re-flash your rom, resinstall Magisk+LSPosed and see what exactly is causing this problem...
